### DISCUSSION TITLE
A Default script to trigger other builds within codemagic

### DISCUSSION THREAD

**Discussion by Reedyuk** [codemagic-ci-cd/codemagic-docs#2688] [Ideas]
### Use case

```Text
At the moment I use a curl script to trigger other codemagic workflows, it would be nice to have an out of the box default script which just takes the app id, workflow name, branch, custom vars.

ios app workflow submits app to app store and triggers UI automation tests
If ui automation tests pass, workflow triggers another workflow to submit app to review

This is all performed with alot of boilerplate code.
```

### Idea

```Text
All CI images have a .sh script which could be used.

- &triggerUIAutomationTests
  name: Triggers UI Automation tests
  script: ./triggerCodeMagic.sh 123 ui_automation_tests develop

Something a bit like this.
```

**Comment by mikhail-tokarev** [comment]
ha I like this optimisation to submit to App Store before running automation tests. I guess you have them a lot, right?)
I'll keep this request open to see if there is more traction. 

One way we are thinking about this problem, is to create staged workflows with option to run stages in parallel, and define triggering condition (e.g. wait all previous stages to finish). Does it sound like a good solution for you?

**Reply by Reedyuk** [reply]
Yes that would be cool, but I would highlight that I want to be able to trigger a workflow cross apps 🤯
We have a ui automation repo separate from our apps.

**Reply by mikhail-tokarev** [reply]
yeah I see... technically you can clone 2nd repo in this case, but probably not the best solution

**Reply by opanitch** [reply]
I would LOVE parallelization!  being able to run `build`, `test`, `analyze` at the same time allows to fail faster!  Any updates on that?
